### PR TITLE
[MySQL] Support Geometry

### PR DESCRIPTION
**Pull request by Tang8330**
## Change

* Adding support for `Geometry`
* Fix byte order bug with Point

### CODE REVIEW

File: lib/mysql/schema/convert.go [hunk 1]
```diff
@@ -126,8 +126,7 @@ func ConvertValue(value any, colType DataType) (any, error) {
 		}
 
 		// The first four bytes are the SRID.
-
-		if byteOrder := bytes[5]; byteOrder != 1 {
+		if byteOrder := bytes[4]; byteOrder != 1 {
```
Comment (Tang8330) [inline lib/mysql/schema/convert.go:129]: This was wrong, didn't account for the zero index. https://go.dev/play/p/LzrwJJScjVE

Comment (Tang8330) [reply lib/mysql/schema/convert.go:129]: @nathan-artie FYI

File: lib/mysql/schema/convert.go [hunk 2]
```diff
@@ -139,6 +138,31 @@ func ConvertValue(value any, colType DataType) (any, error) {
...
+		}
+
+		if len(bytes) != 25 {
+			return nil, fmt.Errorf("expected []byte with length 25, length is %d", len(bytes))
+		}
+
+		var byteOrder binary.ByteOrder
+		switch bytes[4] {
+		case 0:
+			byteOrder = binary.BigEndian
```
Comment (nathan-artie) [inline lib/mysql/schema/convert.go:154]: Is this ever big-endian?

File: lib/mysql/schema/convert.go (hunk 1 above)
Comment (nathan-artie) [reply lib/mysql/schema/convert.go:129]: Good catch!

**Review by nathan-artie — APPROVED** [review]

File: lib/mysql/schema/convert.go (hunk 2 above)
Comment (nathan-artie) [reply lib/mysql/schema/convert.go:154]: I don't think that matters though as MySQL always uses little-endian to store wkbs: https://dev.mysql.com/doc/refman/8.4/en/gis-data-formats.html#gis-wkb-format:~:text=The%20byte%2Dorder%20indicator%20byte%20is%201%20because%20MySQL%20stores%20geometries%20as%20little%2Dendian%20values.

**Pull request merged**